### PR TITLE
Handle kvm system events on aarch64

### PR DESCRIPTION
**Pull request by AlexandruCihodaru**
Signed-off-by: Alexandru Cihodaru <cihodar@amazon.com>

## Reason for This PR

Kvm system events were not handled and they caused `unexpected`
errors.

## Description of Changes

Added handler for KVM_SYSTEM_EVENT_SHUTDOWN
KVM_SYSTEM_EVENT_CRASH and KVM_SYSTEM_EVENT_RESET
on aarch64

- [ ] This functionality can be added in [`rust-vmm`](https://github.com/rust-vmm).

## License Acceptance

By submitting this pull request, I confirm that my contribution is made under
the terms of the Apache 2.0 license.

## PR Checklist
- [x] All commits in this PR are signed (`git commit -s`).
- [x] The reason for this PR is clearly provided (issue no. or explanation).
- [x] The description of changes is clear and encompassing.
- [x] Any required documentation changes (code and docs) are included in this PR.
- [x] Any newly added `unsafe` code is properly documented.
- [x] Any API changes are reflected in `firecracker/swagger.yaml`.
- [x] Any user-facing changes are mentioned in `CHANGELOG.md`.
- [x] All added/changed functionality is tested.

### CODE REVIEW

File: src/vmm/src/vstate/vcpu/mod.rs [hunk 1]
```diff
@@ -473,6 +473,38 @@ impl Vcpu {
                         VcpuExit::InternalError
                     )))
                 }
+                VcpuExit::SystemEvent(event_type, event_flag) => {
+                    METRICS.vcpu.failures.inc();
+                    match event_type {
+                        KVM_SYSTEM_EVENT_RESET | KVM_SYSTEM_EVENT_SHUTDOWN => {
+                            info!(
+                                "Received KVM_SYSTEM_EVENT: type: {}, event: {}",
```
Comment (alindima) [inline src/vmm/src/vstate/vcpu/mod.rs:481]:
```suggestion
                                "Received KVM_SYSTEM_EVENT: type: {}, flag: {}",
```

File: tests/integration_tests/build/test_coverage.py [hunk 2]
```diff
@@ -23,7 +23,7 @@
 # this contains the frequency while on AMD it does not.
 # Checkout the cpuid crate. In the future other
 # differences may appear.
-COVERAGE_DICT = {"Intel": 85.35, "AMD": 84.63, "ARM": 82.87}
+COVERAGE_DICT = {"Intel": 85.25, "AMD": 84.63, "ARM": 82.87}
```
Comment (alindima) [inline tests/integration_tests/build/test_coverage.py:26]: I see the coverage drop. Are there any rust tests that cover the VCPU emulation loop? Could we add these new event types to a test?

File: src/vmm/src/vstate/vcpu/mod.rs [hunk 3]
```diff
@@ -473,6 +473,38 @@ impl Vcpu {
                         VcpuExit::InternalError
                     )))
                 }
+                VcpuExit::SystemEvent(event_type, event_flag) => {
```
Comment (serban300) [inline src/vmm/src/vstate/vcpu/mod.rs:476]: Nit: `event_flags` maybe ? Looks like there might be more.

**Review by serban300 — APPROVED** [review]

File: tests/integration_tests/build/test_coverage.py (hunk 2 above)
Comment (dianpopa) [reply tests/integration_tests/build/test_coverage.py]: There are no unit tests as far as I know that exercise the emulation loop in Firecracker (would be out of the scope fo this PR to add the framework for that here). What we can do as far as testing this functionality is make use of this [integration test](https://github.com/firecracker-microvm/firecracker/blob/master/tests/integration_tests/functional/test_shut_down.py#L58) to test the expected message.

**Comment by sandreim** [comment]
Looks good, what it would take to implement an integration test ?

**Review by serban300 — DISMISSED** [review]

**Review by alindima — APPROVED** [review]

**Comment by dianpopa** [comment]
> Looks good, what it would take to implement an integration test ?

We could test that [this integration test](https://github.com/firecracker-microvm/firecracker/blob/master/tests/integration_tests/functional/test_shut_down.py#L58) does not output the old message anymore.

**Comment by dianpopa** [comment]
Modified an already existent integration test to check behavior of Firecracker when KVM_SYSTEM_EVENT_RESET is sent. @alindima @serban300 @sandreim please review! Thanks!

**Review by alindima — DISMISSED** [review]

**Review by alindima — APPROVED** [review]

**Review by serban300 — APPROVED** [review]

**Pull request merged**